src/dbnode/storage/namespace.go

```diff
@@ -1778,7 +1774,7 @@ func (n *dbNamespace) aggregateTiles(
 		}
 
 		shardProcessedTileCount, err := targetShard.AggregateTiles(
-			sourceNs.ID(), sourceShard.ID(), blockReaders, writer, sourceBlockVolumes, opts, nsCtx.Schema)
+			sourceNs.ID(), n, sourceShard.ID(), blockReaders, writer, sourceBlockVolumes, opts)
```

Comment:
Should this be done under RLock if we're passing in `n`?

Reply:
The method called accepts `n` as a `storage.Namespace` interface which only exposes public methods of `dbNamespace`. These public methods have locking when they access mutable fields of `dbNamespace` (eg. `dbNamespace.Schema()`. So I guess this should be fine?